dipy/reconst/dki.py

```diff
@@ -175,6 +522,382 @@ def _directional_kurtosis(dt, MD, kt, V, min_diffusivity=0, min_kurtosis=-1):
+    # Error is defined as three order of magnitude larger than system's epslon
+    er = np.finfo(a.ravel()[0]).eps * 1e3
+
+    # Initialize G2
+    G2 = np.zeros(a.shape)
+
+    # Only computes G2 in voxels that have all eigenvalues larger than zero
+    cond0 = _positive_evals(a, b, c)
+
+    # Apply formula for non problematic plaussible cases, i.e. b!=c
+    cond1 = np.logical_and(cond0, (abs(b - c) > er))
+    if np.sum(cond1) != 0:
+        L1 = a[cond1]
+        L2 = b[cond1]
+        L3 = c[cond1]
+        G2[cond1] = \
+            (L1+L2+L3)**2 / (3 * (L2-L3)**2) * ((L2+L3) / np.sqrt(L2*L3) - 2)
+
+    # Resolve possible sigularity b==c
+    cond2 = np.logical_and(cond0, abs(b - c) < er)
+    if np.sum(cond2) != 0:
+        L1 = a[cond2]
+        L2 = b[cond2]
+        G2[cond2] = (L1 + 2.*L2)**2 / (12.*L2**2)
+
+    return G2
+
+
+def radial_kurtosis(dki_params, min_kurtosis=0, max_kurtosis=3):
+    r""" Radial Kurtosis (RK) of a diffusion kurtosis tensor.
+
+    Parameters
+    ----------
+    dki_params : ndarray (x, y, z, 27) or (n, 27)
+        All parameters estimated from the diffusion kurtosis model.
+        Parameters are ordered as follow:
+            1) Three diffusion tensor's eingenvalues
+            2) Three lines of the eigenvector matrix each containing the first,
+               second and third coordinates of the eigenvector
+            3) Fifteen elements of the kurtosis tensor
+    min_kurtosis : float (optional)
+        To keep kurtosis values within a plausible biophysical range, radial
+        kurtosis values that are smaller than `min_kurtosis` are replaced with
+        `min_kurtosis`. defaut = 0
+    max_kurtosis : float (optional)
+        To keep kurtosis values within a plausible biophysical range, radial
+        kurtosis values that are larger than `max_kurtosis` are replaced with
+        `max_kurtosis`. defaut = 3
+
+    Returns
+    -------
+    rk : array
+        Calculated RK.
+
+    Notes
+    --------
+    RK is calculated with the following equation:
+    .. math::
+        K_{\bot} = G_1(\lambda_1,\lambda_2,\lambda_3)\hat{W}_{2222} +
+                   G_1(\lambda_1,\lambda_3,\lambda_2)\hat{W}_{3333} +
+                   G_2(\lambda_1,\lambda_2,\lambda_3)\hat{W}_{2233}
+
+    where:
+    .. math::
+        G_1(\lambda_1,\lambda_2,\lambda_3)=
+        \frac{(\lambda_1+\lambda_2+\lambda_3)^2}{18\lambda_2(\lambda_2-
+        \lambda_3)} \left (2\lambda_2 +
+        \frac{\lambda_3^2-3\lambda_2\lambda_3}{\sqrt{\lambda_2\lambda_3}}
+        \right)
+
+    and
+
+    .. math::
+        G_2(\lambda_1,\lambda_2,\lambda_3)=
+        \frac{(\lambda_1+\lambda_2+\lambda_3)^2}{(\lambda_2-\lambda_3)^2}
+        \left ( \frac{\lambda_2+\lambda_3}{\sqrt{\lambda_2\lambda_3}}-2\right )
+    """
+    # Flat parameters. For numpy versions more recent than 1.6.0, this step
+    # isn't required
+    outshape = dki_params.shape[:-1]
+    dki_params = dki_params.reshape((-1, dki_params.shape[-1]))
```

Comment:
I think I cannot use here flatten or ravel, because these creates 1D arrays. Instead, I want to create a 2D array with dimension of (N, 27) where N is the number of voxels and 27 the number of parameters estimated from DKI.